starknet/.gitignore

```diff
@@ -0,0 +1 @@
+**/rust/
```

Comment:
Why the changes in the .gitignore files?

Reply:
Good question, these are rust untracked files - I belive it's starknet/compiler/* that appears out of nowhere after I compiled the all thing

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	starknet/_gitignore
	starknet/compiler/rust/Cargo.lock
	starknet/compiler/rust/target/
	starknet/rust/
```

Reply:
these files shouldn't be tracked because we have .gitignore in /starknet/compiler/rust dir

Reply:
you removed it in this PR